### ISSUE TITLE
Improve the GIS overlay analysis if it's a point exposure with raster hazard

### ISSUE THREAD

**Issue by Gustry** [inasafe/inasafe#3469]
### Problem
For now, we reclassify, polygonize every hazard raster.

We can improve the workflow if it's a raster hazard and point exposure by doing a pixel lookup. It will improve a lot the processing time.

**Comment by Charlotte-Morgan** [comment]
is this still something you plan to do @Gustry

**Comment by Gustry** [comment]
Yes if I have some spare time or if you make it a priority for the next release ;-)
We can decrease a lot the processing time by doing this tweak.

**Comment by timlinux** [comment]
See also #4507 and #4262 - this same logic can be used for calculating places affected

**Comment by timlinux** [comment]
@gustry to make a helper function that takes a hazard value and returns its class.

**Comment by Gustry** [comment]
To retrieve the current hazard classification which has to be used (according to the exposure type):
```
        classification_key = active_classification(
            layer.keywords, exposure_key)
        thresholds = active_thresholds_value_maps(layer.keywords, exposure_key)
```

Then there is a helper: `reclassified_value = reclassify_value(source_value, thresholds)`

@ivanbusthomi Does it work for you? I'm not sure what you need, so just ping me ;-)
